test/e2e/external_gateways.go

```diff
@@ -1191,13 +1201,13 @@ func setupGatewayPods(f *framework.Framework, nodes *v1.NodeList, pod1, pod2, ns
 		if addressesv6.srcPodIP != "" && addressesv6.nodeIP != "" {
 			networkIPs = fmt.Sprintf("\"%s\", \"%s\"", addressesv4.gatewayIPs[i], addressesv6.gatewayIPs[i])
 		}
-		annotatePodForGateway(gwPod, f.Namespace.Name, networkIPs, bfd)
+		annotatePodForGateway(gwPod, ns, f.Namespace.Name, networkIPs, bfd)
 	}
 
 	return gwPods
 }
 
-func cleanExGWContainersAndPods(clientSet kubernetes.Interface, gwContainers, gwPods []string, ns string, addressesv4, addressesv6 gatewayTestIPs) {
```

Comment:
do we not need to delete the pods anymore? Is that because the namespace is deleted? If so where is that happening?

Reply:
right, it's a side effect of creating a namespace through the framework - inside the create they also `f.AddNamespacesToDelete(ns)`